### PR TITLE
Modify client to notice schema changes v2

### PR DESCRIPTION
**Pull request by NikkyXO**

### CODE REVIEW

**Comment by NikkyXO** [comment]
> 

implemented and test passes @spbnick

File: kcidb/test_db.py [hunk 1]
```diff
@@ -18,6 +19,17 @@ def test_schemas_main():
     assert_executes("", *argv,
                     stdout_re=r"4\.0: 4\.0\n4\.1: 4\.1\n")
 
+def test_reset(clean_database):
+    """Check kcidb-db-reset  works"""
```
Comment (spbnick) [inline kcidb/test_db.py:23]: There's no such command-line tool, and it wouldn't make sense to create one, because `reset()` only affects the client-side state. This is a test for the `Client.reset()` method you just added. Please fix the docstring.

File: kcidb/db/__init__.py [hunk 2]
```diff
@@ -89,6 +87,16 @@ def is_initialized(self):
         """
         return self.driver.is_initialized()
 
+    def reset(self):
+        """
+        Recreates a database driver instance on request
+        using existing database string
```
Comment (spbnick) [inline kcidb/db/__init__.py:93]: This basically describes the implementation and is not very useful for the user. We need more of a conceptual description. Something like "Reset the client's database connection, updating its knowledge of the database schema."

**Review by spbnick — CHANGES_REQUESTED** [review]
The code looks good, thank you @NikkyXO. Please fix the docstrings and we'll merge it :relaxed:

File: kcidb/test_db.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
 import pytest
 import kcidb
 from kcidb.unittest import assert_executes
+from kcidb.db import Client
```
Comment (spbnick) [inline kcidb/test_db.py:11]: Please remove this import, and refer to the `Client` class as `kcidb.db.Client`, like the rest of the code here does.

File: kcidb/test_db.py [hunk 4]
```diff
@@ -18,6 +19,17 @@ def test_schemas_main():
     assert_executes("", *argv,
                     stdout_re=r"4\.0: 4\.0\n4\.1: 4\.1\n")
 
+def test_reset(clean_database):
+    """Check Client.reset  works"""
```
Comment (spbnick) [inline kcidb/test_db.py:23]: Could you add the parens after `Client.reset` to signify it's a function/method (so it becomes `Client.reset()`), and also remove the extra space before "works"? So that the complete docstring becomes `"""Check Client.reset() works"""`?

File: kcidb/db/__init__.py (hunk 2 above)
Comment (spbnick) [reply kcidb/db/__init__.py:93]: Please take care of this request as well, in the corresponding commit you're adding the function.

**Review by spbnick — CHANGES_REQUESTED** [review]
Still a few bits to fix, but we're close. Thank you :relaxed:

File: kcidb/test_db.py [hunk 5]
```diff
@@ -9,6 +9,7 @@
 import kcidb
 from kcidb.unittest import assert_executes
 
+
```
Comment (spbnick) [inline kcidb/test_db.py:12]: Please do not do unrelated whitespace changes. Please remove this empty line.

File: kcidb/test_db.py [hunk 6]
```diff
@@ -18,6 +19,17 @@ def test_schemas_main():
     assert_executes("", *argv,
                     stdout_re=r"4\.0: 4\.0\n4\.1: 4\.1\n")
 
+def test_reset(clean_database):
+    """Check Client.reset() works"""
+
```
Comment (spbnick) [inline kcidb/test_db.py:24]: Could you remove this empty line? We'll be fine without it and the code will be shorter.

**Review by spbnick — CHANGES_REQUESTED** [review]
Looks good, thank @NikkyXO, but please squash these commits into one, as the test doesn't really make sense without the code change, and they should go together.

Apart from the minor requests inline, please make sure CI passes before the next review.

Thank you!

**Comment by spbnick** [comment]
This looks great and would be ready to go in, but you need to rebase it on latest `main` first, as there's a conflict with your previous change, which added storing of the database string. Ask for help with this on Slack, if you need it.

**Comment by NikkyXO** [comment]
> This looks great and would be ready to go in, but you need to rebase it on latest `main` first, as there's a conflict with your previous change, which added storing of the database string. Ask for help with this on Slack, if you need it.

i have rebased with upstream/main after a git fetch and merge @spbnick

**Comment by spbnick** [comment]
Looks great! Thank you, @NikkyXO! Merging.
